tests/esbuild_migration.test.ts

```diff
@@ -0,0 +1,100 @@
+
+describe('ESBuild Migration', () => {
+  /**
+   * The two most prominent import expressions are:
+   *  - require("caniuse-lite/data/regions/"+n+".js")
+   *  - require("caniuse-lite/data/features/"+t+".js")
+   * Both are within browserslist: https://github.com/browserslist/browserslist/blob/main/node.js
+   * And both are not supported by esbuild, neither by our fork nor by upstream.
+   */
+  testMany('caniuse-lite', esbuildConfigs, async (opts) => {
+    await expect(() => zipNode('caniuse-lite', { opts })).rejects.toThrowError('Unknown region name `DE`.')
+  })
+
+  /**
+   * This test covers `require("../chunks/" + __webpack_require__.u(chunkId))`.
+   * arguably one of the most important patterns for us to support.
+   * It's supported by our ESBuild fork, but not by upstream.
+   * This is blocking the migration for now, opened an issue about it:
+   * https://github.com/evanw/esbuild/issues/3328
+   * In the meantime, we found a workaround.
+   */
+  testMany('webpack chunks', esbuildConfigs, async (options) => {
+    const fixtureName = 'webpack-chunks'
+    const opts = merge(options, {
+      basePath: join(FIXTURES_DIR, fixtureName),
+    })
+    await zipNode(fixtureName, { opts })
+  })
+
+  /**
+   * This is the same test as above, but with a `.cjs` chunk.
+   * It works on neither our fork nor upstream.
+   */
+  testMany('webpack chunks .cjs', esbuildConfigs, async (options) => {
+    const fixtureName = 'webpack-chunks-cjs'
+    const opts = merge(options, {
+      basePath: join(FIXTURES_DIR, fixtureName),
+    })
+    await expect(() => zipNode(fixtureName, { opts })).rejects.toThrowError()
+  })
+
+  /**
+   * This is the same test as above, but with a `.json` chunk.
+   */
+  testMany('webpack chunks .json', esbuildConfigs, async (options) => {
+    const fixtureName = 'webpack-chunks-json'
+    const opts = merge(options, {
+      basePath: join(FIXTURES_DIR, fixtureName),
+    })
+    await zipNode(fixtureName, { opts })
+  })
+
+  /**
+   * This is the same test as above, but with a binary file.
+   * It fails on our fork and upstream.
+   */
+  testMany('webpack chunks .node', esbuildConfigs, async (options) => {
+    const fixtureName = 'glob-require-native-module'
+    const opts = merge(options, {
+      basePath: join(FIXTURES_DIR, fixtureName),
+    })
+    await expect(() => zipNode(fixtureName, { opts })).rejects.toThrowError()
+  })
+
+  /**
+   * This test covers `require(`cardinal${REQUIRE_TERMINATOR}`)`.
+   */
+  testMany('cardinal require terminator', esbuildConfigs, async (opts) => {
+    const {
+      files: [{ unzipPath, entryFilename }],
+    } = await zipNode('cardinal-require-terminator', { opts })
+    const res = await importFunctionFile(`${unzipPath}/${entryFilename}`)
+    expect(res).toEqual('Cardinal is unavailable!')
+  })
+
+  /**
+   * We're seeing one site do `require(`../../data/${req.params.foo}/${req.params.bar}`)`.
+   * It's not supported on our fork, but it is on upstream!
```

Comment:
The fact that it's supported by upstream means that this test will fail in CI. If you see that, it's expected.